tests/core/test_model.py

```diff
@@ -4445,3 +4445,42 @@ def test_incremental_by_partition(sushi_context, assert_exp_eq):
             """
         )
         load_sql_based_model(expressions)
+
+
+@pytest.mark.parametrize(
+    ["model_def", "path", "assertion"],
+    [
+        [
+            """dialect snowflake,""",
+            """models/test_schema/test_model.sql,""",
+            "test_schema.test_model",
+        ],
+        [
+            """dialect snowflake,""",
+            """models/test_model.sql,""",
+            "test_model",
+        ],
+        [
+            """dialect snowflake,""",
+            """models/catalog1/catalog2/test_schema/test_model.sql,""",
+            "catalog1.catalog2.test_schema.test_model",
```

Comment:
Is this a correct name?

Reply:
no, this is not a legal name

Reply:
Logic was adapted to go only 3 levels deep as discussed, so this test was adapted accordingly.